tools/development/parser/grammar.y

```diff
@@ -0,0 +1,1223 @@
+%{
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * This is imported from GStreamer and altered to parse GST-Pipeline
+ */
+
+#include <glib-object.h>
+#include <glib.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include "types.h"
+
+/* All error messages in this file are user-visible and need to be translated.
+ * Don't start the message with a capital, and don't end them with a period,
+ * as they will be presented inside a sentence/error.
+ */
+
+#define YYERROR_VERBOSE 1
+
+#define YYENABLE_NLS 0
+
+#ifndef YYLTYPE_IS_TRIVIAL
+#define YYLTYPE_IS_TRIVIAL 0
+#endif
+
+/*******************************************************************************************
+*** define SET_ERROR macro/function
+*******************************************************************************************/
+#ifdef G_HAVE_ISO_VARARGS
+
+#  define SET_ERROR(error, type, ...) \
+G_STMT_START { \
+  GST_CAT_ERROR (GST_CAT_PIPELINE, __VA_ARGS__); \
+  if ((error) && !*(error)) { \
+    g_set_error ((error), GST_PARSE_ERROR, (type), __VA_ARGS__); \
+  } \
+} G_STMT_END
+
+#elif defined(G_HAVE_GNUC_VARARGS)
+
+#  define SET_ERROR(error, type, args...) \
+G_STMT_START { \
+  GST_CAT_ERROR (GST_CAT_PIPELINE, args ); \
+  if ((error) && !*(error)) { \
+    g_set_error ((error), GST_PARSE_ERROR, (type), args ); \
+  } \
+} G_STMT_END
+
+#else
+
+static inline void
+SET_ERROR (GError **error, gint type, const char *format, ...)
+{
+  if (error) {
+    if (*error) {
+      g_warning ("error while parsing");
+    } else {
+      va_list varargs;
+      char *string;
+
+      va_start (varargs, format);
+      string = g_strdup_vprintf (format, varargs);
+      va_end (varargs);
+
+      g_set_error (error, GST_PARSE_ERROR, type, string);
+
+      g_free (string);
+    }
+  }
+}
+
+#endif /* G_HAVE_ISO_VARARGS */
+
+/*** define YYPRINTF macro/function if we're debugging */
+
+/* bison 1.35 calls this macro with side effects, we need to make sure the
+   side effects work - crappy bison */
+
+#ifndef GST_DISABLE_GST_DEBUG
+#  define YYDEBUG 1
+
+#  ifdef G_HAVE_ISO_VARARGS
```

Comment:
Q) is it intended indentation like a code block?

Reply:
That part is from GStreamer Parser without touching the original code.